maildir.go

```diff
@@ -127,18 +236,24 @@ func (d Dir) Unseen() ([]string, error) {
 			// that case.
 			key, _, _ := strings.Cut(n, string(separator))
 			info := "2,"
+			newBasename := key + string(separator) + info
 
 			err = os.Rename(filepath.Join(string(d), "new", n),
-				filepath.Join(string(d), "cur", key+string(separator)+info))
+				filepath.Join(string(d), "cur", newBasename))
+			if err != nil {
+				return msgs, err
+			}
+
+			msg, err := d.newMessage(filepath.Join(string(d), "cur"), newBasename)
 			if err != nil {
-				return keys, err
+				panic(err) // unreachable
 			}
 
-			keys = append(keys, key)
+			msgs = append(msgs, *msg)
```

Comment:
Why dereferencing the pointer here?

Reply:
Wouldn't it be better to return a list of pointers?

Reply:
Yup, that's the first TODO in the PR description.

Generally using values directly is a tad better when dealing with a large slice, because it saves one allocation per slice entry. But here we expose methods which mutate the messages, so it's a footgun in `for` loops.